posts/views.py

```diff
@@ -32,3 +34,29 @@ def list(self, request) -> Response:
 
         serializer = PostSerializer(queryset, many=True)
         return Response(serializer.data)
+
+
+class LikeDislikeViewSet(viewsets.ModelViewSet):
+    permission_classes = (IsAuthenticated,)
+
+    def post(self, request, pk: int):
+        response = "Setted"
+        is_like = self.request.GET.get("like") == "true"
+        if not LikeDislike.objects.filter(post_id=pk).filter(user=request.user):
+            LikeDislike.objects.create(user=request.user, post_id=pk, is_like=is_like)
+        else:
+            like = LikeDislike.objects.get(user=request.user, post_id=pk)
+            if is_like != like.is_like:
+                like.is_like = not like.is_like
+                like.save()
+            else:
+                response = "Like or dislike already exists"
```

Comment:
Может я тут ошибся где-то, но смысл такой:
```suggestion
            LikeDislike.objects.update_or_create(user=request.user, post_id=pk, is_like=is_like)
```
Посмотри в доках как это написать правильно